app/circuits-interactions.js

```diff
@@ -0,0 +1,136 @@
+const { createMerkleTree } = require('./utils/merkletree')
+const { stringifyBigInts, unstringifyBigInts } = require('./utils/helpers')
+const { randomPrivateKey, privateToPublicKey, signAndEncrypt, ecdh } = require('./utils/crypto')
+
+const provingKey = require('./circuits/proving_key.json')
+const verificationKey = require('./circuits/verification_key.json')
+const circuitDef = require('./circuits/circuit.json')
+
+const { Circuit, groth } = require('snarkjs')
+// const { mimc7 } = require('circomlib')
+
+const zkSnark = groth
+
+// Command and State Tree
+const cmdTree = createMerkleTree(4, 0n)
+const stateTree = createMerkleTree(4, 0n)
+
+// Coordinator keys
+const coordinatorSecretKey = randomPrivateKey()
+const coordinatorPublicKey = privateToPublicKey(coordinatorSecretKey)
+
+// Create 3 users (for registration)
+const user1SecretKey = randomPrivateKey()
+const user1PublicKey = privateToPublicKey(user1SecretKey)
+const user1Message = [...user1PublicKey, 0n]
+const user1EncryptedMsg = signAndEncrypt(
+  user1Message,
+  user1SecretKey,
+  user1SecretKey,
```

Comment:
Why do you pass user1SecretKey twice?

Reply:
That's because signing and encrypting are using two different keys. We need to somehow be able to generate the ecdh key, so one public key will always be exposed